tsconfig.base.json

```diff
@@ -8,9 +8,10 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "target": "es2015",
+    "allowUnreachableCode": false,
+    "target": "es2017",
     "typeRoots": ["node_modules/@types"],
-    "lib": ["es2017", "dom", "es6", "es2015.iterable"],
+    "lib": ["es2017", "dom"],
```

Comment:
What is happening here?

Reply:
The target (previously `es2015`) is supplemented by the `lib` option which is used to indicate what extra JS API’s/features the compiled project uses on top of `es2015`. So if we supply polyfills for some or all `es2017`-specific features we can indicate that in the `lib` config. 

But I can only think of use cases where we want to compile to a lower target like `es2015` to support older browsers, but then supply polyfills for some features, so the runtime can be es2017. And to silence the compiler we then specify what functionality we ensure ourselves to deliver through the `lib` config. 

Since our previous target was `es2015` (which is also es6), there should never have been a need for these, as we already have a target that supports these features. 

In conclusion I cant see anything wrong with removing this. It is overwritten for our published package, so it is only relevant for the compiled cookbook code, and the core library.

https://www.typescriptlang.org/tsconfig#lib
https://stackoverflow.com/a/42097465
https://www.meziantou.net/which-version-of-ecmascript-should-i-use-in-the-typescript-configuration.htm

Reply:
We should consider to bump the target of [tsconfig.lib.json] (https://github.com/kirbydesign/designsystem/blob/master/libs/designsystem/tsconfig.lib.json) (published angular lib) to `es2017` since we are already 'promising' this and more by setting the 'lib' to `es2018`. To my knowledge we are not supplying polyfills to close that gap.